ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -141,7 +164,29 @@
    */
   public BlobId(short version, BlobIdType type, byte datacenterId, short accountId, short containerId,
```

Comment:
this constructor should take in the `BlobDataType`. There should be no reason to construct a new blob id with a null value for the `BlobDataType` - correct? (or in other words, what piece of production code would use this constructor?)

Reply:
I don't disagree, but this constructor is called in 63 places.  Should we change all of them?  It seems easier to maintain to use fluent constructors and setters, eg. new BlobId().version(5).type(NATIVE).etc()

Reply:
Agreed on having a builder sort of construction on this. We should do this in a follow up PR.